test/unit/forms/declaration/DepartureTransportSpec.scala

```diff
@@ -50,94 +51,233 @@ class DepartureTransportSpec extends FormSpec {
 
     "has errors" when {
 
-      "mandatory field are empty" in {
+      val transportType = IMOShipIDNumber
+      val transportTypeField = DepartureTransport.meansOfTransportOnDepartureTypeKey
+      val idNumberField = s"meansOfTransportOnDepartureIDNumber_$IMOShipIDNumber"
+
+      "transport type not selected" in {
+
+        val incorrectForm =
+          Map(transportTypeField -> "", idNumberField -> "")
+
+        val result = form.bind(incorrectForm)
+        val errorKeys = result.errors.map(_.key)
+        val errorMessages = result.errors.map(_.message)
+
+        errorKeys must be(List(transportTypeField))
+        errorMessages must be(List("declaration.transportInformation.meansOfTransport.departure.error.empty"))
+      }
+
+      "transport id not provided" in {
+
+        val incorrectForm =
+          Map(transportTypeField -> IMOShipIDNumber, idNumberField -> "")
+
+        val result = form.bind(incorrectForm)
+        val errorKeys = result.errors.map(_.key)
+        val errorMessages = result.errors.map(_.message)
+
+        errorKeys must be(List(idNumberField))
+        errorMessages must be(List("declaration.transportInformation.meansOfTransport.reference.error.empty"))
+      }
+
+      "fields are incorrect" in {
+
+        val incorrectForm = Map(transportTypeField -> "incorrect", idNumberField -> "correct")
```

Comment:
The value for `idNumberField` is "correct" while test case says "fields are incorrect". Is it just because this field accepts any string except an empty one? And so one can't really make it incorrect?

Reply:
Just noticed there is a test "means of transport on departure id number is too long", so I'd recommend giving field `idNumberField` some invalid value.